exporter/exporter.go

```diff
@@ -594,6 +596,7 @@ func NewRedisExporter(uri string, opts Options) (*Exporter, error) {
 		"stream_radix_tree_nodes":                            {txt: `Radix tree nodes count`, lbls: []string{"db", "stream"}},
 		"up":                                                 {txt: "Information about the Redis instance"},
 	} {
+		desc.lbls = append(desc.lbls, "instance_role")
```

Comment:
We can add this label only if option is enabled, but afaik if value is not provided, label doesn't exist in metrics. 

Please let me know if it's better to add `if` here.

Reply:
Never-mind, tests in CI proved that label shouldn't be added if not used.

It was causing panics in local test, like:

```log
panic: inconsistent label cardinality: expected 2 label values but got 1 in []string{"set"}

goroutine 39 [running]:
github.com/prometheus/client_golang/prometheus.MustNewConstSummary(...)
```